src/components/DealProposalForm.tsx

```diff
@@ -58,12 +58,14 @@ const FormInput = ({
   accounts,
   selectedAccount,
   onSelectAccount,
+  currentBlock,
 }: {
   dealProposal: InputFields;
   onChange: (dealProposal: InputFields) => void;
   accounts: InjectedAccountWithMeta[];
   selectedAccount: InjectedAccountWithMeta | null;
   onSelectAccount: (account: InjectedAccountWithMeta) => void;
+  currentBlock: number | null;
```

Comment:
Doesn't this trigger a re-render of the whole component every ~6 seconds?

Reply:
Yeah, but I'm not sure we can avoid that.

Reply:
Let's skip it and treat is as `not-worth-it` optimisation.